python/cudf/cudf/core/cut.py

```diff
@@ -0,0 +1,307 @@
+    array([0, 1, 1, 3], dtype=int32)
+    Passing a Series as an input returns a Series with categorical dtype:
+    >>> s = pd.Series(np.array([2, 4, 6, 8, 10]),
+    ...        index=['a', 'b', 'c', 'd', 'e'])
+    >>> pd.cut(s, 3)
+    """
+    left_inclusive = False
+    right_inclusive = True
+    # saving the original input x for use in case its a series
+    orig_x = x
+    old_bins = bins
+
+    if not ordered and labels is None:
+        raise ValueError("'labels' must be provided if 'ordered = False'")
+
+    if duplicates not in ["raise", "drop"]:
+        raise ValueError(
+            "invalid value for 'duplicates' parameter, valid options are: "
+            "raise, drop"
+        )
+
+    if labels is not False:
+        if not (labels is None or is_list_like(labels)):
+            raise ValueError(
+                "Bin labels must either be False, None or passed in as a "
+                "list-like argument"
+            )
+        elif ordered and labels is not None:
+            if len(set(labels)) != len(labels):
+                raise ValueError(
+                    "labels must be unique if ordered=True;"
+                    "pass ordered=False for duplicate labels"
+                )
+
+    # bins can either be an int, sequence of scalars or an intervalIndex
+    if isinstance(bins, Sequence):
+        if len(set(bins)) is not len(bins):
+            if duplicates == "raise":
+                raise ValueError(
+                    f"Bin edges must be unique: {repr(bins)}.\n"
+                    f"You can drop duplicate edges by setting the 'duplicates'"
+                    "kwarg"
+                )
+            elif duplicates == "drop":
+                # get unique values but maintain list dtype
+                bins = list(dict.fromkeys(bins))
+
+    # if bins is an intervalIndex we ignore the value of right
+    if isinstance(bins, (pd.IntervalIndex, cudf.IntervalIndex)):
+        right = bins.closed == "right"
+
+    # create bins if given an int or single scalar
+    if not isinstance(bins, pd.IntervalIndex):
+        if not isinstance(bins, (Sequence)):
+            if isinstance(
+                x, (pd.Series, cudf.Series, np.ndarray, cupy.ndarray)
+            ):
+                # changing to scalars and using sequence to get the bins
+                # because this allows masked arrays from value_counts to
+                # also be able to be handled by cut correctly
+                # pandas by default seems to turn all bins into a float
+                mn = x.min()
+                mx = x.max()
+            else:
+                mn = min(x)
+                mx = max(x)
+            # step = cudf.Scalar((mx - mn) / bins, dtype="float64")
+            bins = np.linspace(mn, mx, bins + 1, endpoint=True)
+            # this is another possible way to calculate bins
+            # bins = sequence(
+            #     size=bins + 1, init=mn.device_value, step=step.device_value
+            # ).values
```

Comment:
Some stale codes

Reply:
Thanks for the catch here Michael :D I'll remove this!